### ISSUE TITLE
running tox in parallel hits a race condition with logger

### ISSUE THREAD

**Issue by mattp-** [tox-dev/tox#1064]
it looks like the instantiation of Session objects empties the log dir at https://github.com/tox-dev/tox/blob/master/src/tox/session.py#L379 when a tox run occurs. when this happens with multiple instances running in parallel, it looks like a race occurs where one instance clobbers another:
```
removing /repo/.tox/log
using tox.ini: /repo/tox.ini
using tox-3.5.2 from /usr/local/lib/python3.6/site-packages/tox/__init__.py
GLOB start: packaging
GLOB sdist-make: /repo/setup.py
  removing /repo/.tox/dist
  /repo$ /usr/bin/python36 /repo/setup.py sdist --formats=zip --dist-dir /repo/.tox/dist >/repo/.tox/log/tox-0.log
  removing /repo/.tox/log
using tox.ini: /repo/tox.ini
using tox-3.5.2 from /usr/local/lib/python3.6/site-packages/tox/__init__.py
lock file /repo/.tox/.package.lock present, will block until released
  removing /repo/.tox/log
using tox.ini: /repo/tox.ini
using tox-3.5.2 from /usr/local/lib/python3.6/site-packages/tox/__init__.py
lock file /repo/.tox/.package.lock present, will block until released
  removing /repo/.tox/log
using tox.ini: /repo/tox.ini
using tox-3.5.2 from /usr/local/lib/python3.6/site-packages/tox/__init__.py
lock file /repo/.tox/.package.lock present, will block until released
  removing /repo/.tox/log
using tox.ini: /repo/tox.ini
using tox-3.5.2 from /usr/local/lib/python3.6/site-packages/tox/__init__.py
lock file /repo/.tox/.package.lock present, will block until released
  removing /repo/.tox/log
using tox.ini: /repo/tox.ini
using tox-3.5.2 from /usr/local/lib/python3.6/site-packages/tox/__init__.py
lock file /repo/.tox/.package.lock present, will block until released
  removing /repo/.tox/log
using tox.ini: /repo/tox.ini
using tox-3.5.2 from /usr/local/lib/python3.6/site-packages/tox/__init__.py
lock file /repo/.tox/.package.lock present, will block until released
  removing /repo/.tox/log
using tox.ini: /repo/tox.ini
using tox-3.5.2 from /usr/local/lib/python3.6/site-packages/tox/__init__.py
lock file /repo/.tox/.package.lock present, will block until released
GLOB finish: packaging after 0.93 seconds
GLOB start: packaging
GLOB sdist-make: /repo/setup.py
  removing /repo/.tox/dist
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/py/_error.py", line 66, in checked_call
    return func(*args, **kwargs)
FileNotFoundError: [Errno 2] No such file or directory: '/repo/.tox/log/tox-0.log'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/tox", line 11, in <module>
    sys.exit(cmdline())
  File "/usr/local/lib/python3.6/site-packages/tox/session.py", line 42, in cmdline
    main(args)
  File "/usr/local/lib/python3.6/site-packages/tox/session.py", line 49, in main
    retcode = build_session(config).runcommand()
  File "/usr/local/lib/python3.6/site-packages/tox/session.py", line 446, in runcommand
    return self.subcommand_test()
  File "/usr/local/lib/python3.6/site-packages/tox/session.py", line 566, in subcommand_test
    venv.package = self.hook.tox_package(session=self, venv=venv)
  File "/usr/local/lib/python3.6/site-packages/pluggy/__init__.py", line 617, in __call__
    return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
  File "/usr/local/lib/python3.6/site-packages/pluggy/__init__.py", line 222, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/usr/local/lib/python3.6/site-packages/pluggy/__init__.py", line 216, in <lambda>
    firstresult=hook.spec_opts.get('firstresult'),
  File "/usr/local/lib/python3.6/site-packages/pluggy/callers.py", line 201, in _multicall
    return outcome.get_result()
  File "/usr/local/lib/python3.6/site-packages/pluggy/callers.py", line 76, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/usr/local/lib/python3.6/site-packages/pluggy/callers.py", line 180, in _multicall
    res = hook_impl.function(*args)
  File "/usr/local/lib/python3.6/site-packages/tox/package.py", line 23, in tox_package
    session.package, session.dist = get_package(session)
  File "/usr/local/lib/python3.6/site-packages/tox/package.py", line 43, in get_package
    package = acquire_package(config, report, session)
  File "/usr/local/lib/python3.6/site-packages/tox/package.py", line 97, in acquire_package
    path = build_package(config, report, session)
  File "/usr/local/lib/python3.6/site-packages/tox/package.py", line 124, in build_package
    return make_sdist_legacy(report, config, session)
  File "/usr/local/lib/python3.6/site-packages/tox/package.py", line 149, in make_sdist_legacy
    returnout=True,
  File "/usr/local/lib/python3.6/site-packages/tox/session.py", line 239, in popen
    out = outpath.read()
  File "/usr/local/lib/python3.6/site-packages/py/_path/common.py", line 170, in read
    with self.open(mode) as f:
  File "/usr/local/lib/python3.6/site-packages/py/_path/local.py", line 361, in open
    return py.error.checked_call(open, self.strpath, mode)
  File "/usr/local/lib/python3.6/site-packages/py/_error.py", line 86, in checked_call
    raise cls("%s%r" % (func.__name__, args))
py.error.ENOENT: [No such file or directory]: open('/repo/.tox/log/tox-0.log', 'r')
  /repo$ /usr/bin/python36 /repo/setup.py sdist --formats=zip --dist-dir /repo/.tox/dist >/repo/.tox/log/tox-0.log
actionid: tox
msg: packaging
cmdargs: '/usr/bin/python36 /repo/setup.py sdist --formats=zip --dist-dir /repo/.tox/dist'
```
after each instance gets past this point it looks like the existing file locking used for the sdist process works as expected, so that part is fine in parallel.

**Comment by mattp-** [comment]
oops, looks like this was already covered in https://github.com/tox-dev/tox/issues/641 . closing this

**Comment by gaborbernat** [comment]
@mattp- note if you use isalated build this error will not manifest 👍

**Comment by mattp-** [comment]
@gaborbernat I was actually trying that (well, trying to make sense of pep517/518 really). Maybe you can lend some insight, is there any way to make the isolated build infra use an existing setup.py? I have a fair amount of customization in my setup.py, which seems to ignored even if you specify ```build-backend = "setuptools.build_meta"``` in pyproject.toml. likewise  for requirements.txt / dependency_links for git based repos I couldn't quite figure out. much thanks either way :)

**Comment by gaborbernat** [comment]
I don't think ``requirements.txt`` files are supported inside pyproject.toml. 👍

**Comment by mattp-** [comment]
in case someone encounters the same confusion I had, by default the requirements to _build_ your package are inherited from the base testenv target in tox as well as what is defined in pyproject.toml. If you are using requirements.txt this won't work, but also is not needed as a build time dependency. To short circuit that the newly added https://tox.readthedocs.io/en/latest/config.html#conf-isolated_build_env can be used to override doing something like 
```[testenv:.package]
deps =
```
